src/System.IO.Pipelines/ReadableBuffer.cs

```diff
@@ -57,12 +57,13 @@ internal ReadableBuffer(ReadCursor start, ReadCursor end)
         {
             _start = start;
             _end = end;
+            _length = -1;
+
             if (!end.IsEnd && !end.GreaterOrEqual(start))
             {
-                throw new ArgumentException("End should be greater or equal to start");
+                ThrowHelper.ThrowArgumentException_ReadableBufferCtor();
             }
-            start.TryGetBuffer(end, out _first, out start);
-            _length = -1;
```

Comment:
Don't know if it would; but setting value after function call prevents a tail jump; as it needs to return to set the variable. So set before the call as its uneffected by the result.